static/js/containers/CreatePostPage.js

```diff
@@ -174,9 +183,20 @@ class CreatePostPage extends React.Component<CreatePostPageProps> {
       const { postType, title, url, text } = postForm.value
       const isText = isTextTabSelected(postType, channel)
       const data: CreatePostPayload = isText ? { title, text } : { title, url }
-      dispatch(actions.posts.post(channelName, data)).then(post => {
-        history.push(postDetailURL(channelName, post.id, post.slug))
-      })
+      // $FlowFixMe
+      dispatch(actions.posts.post(channelName, data))
+        .then(post => {
+          history.push(postDetailURL(channelName, post.id, post.slug))
+        })
+        .catch(() => {
+          dispatch(
+            setBannerMessage(
+              `Something went wrong creating your post. Please try again or contact us at ${
+                SETTINGS.support_email
+              }`
+            )
+          )
+        })
```

Comment:
this promise handling code might be a little more readable if written with the `async / await` sugar

Reply:
with `async / await` you can avoid any of the promise chaining type stuff (with `.then`, `.catch`, etc) and write code that just looks synchronous. to replace `.catch` you can do

```js
try {
  const post = await dispatch(actions.posts.post(
    channelName, data
  ))
  history.push(postDetailURL(channelName, post.id, post.slug))
} catch {
  dispatch(
    setBannerMessage(
      `Something went wrong creating your post. Please try again or contact us at ${
        SETTINGS.support_email
      }`
    )
  )
}
```

this is basically equivalent to the `.catch` stuff. I find using `await` ends up generally being a lot more readable than working with raw promises.